src/include/device.h

```diff
@@ -11,7 +11,9 @@
 
 #include "nccl.h"
 #include "rccl_float8.h"
-#include <hip/hip_bfloat16.h>
+// #include <hip/hip_bfloat16.h>
+#include <hip/hip_bf16.h>
```

Comment:
when was this header introduced? can you gate it depending on the header's availability or ROCm version. 

One approach would be this

```suggestion
#if __has_include(<hip/hip_bf16.h>)
  #include <hip/hip_bf16.h>
#else
  #include <hip/hip_bfloat16.h>
#endif

```
but ROCm version should be the way to go

Reply:
This probably needs to be hip version or the mechanism that Mustafa recommends here.  ROCm version strings are not currently monotonic with ROCK.